### ISSUE TITLE
KeePassDroid corrups database when removing entry

### ISSUE THREAD

**Issue by smvicente** [bpellin/keepassdroid#404]
I've noticed the DB gets corrupted every time I remove an entry. This happens at least with v1 "kdb" ones.

**Comment by smvicente** [comment]
Ok I think this might be a duplicate from #375. I'm on Pixel 2XL, and it only happens when removing an entry, not when creating one.

**Comment by bpellin** [comment]
I haven't been able to replicate this. Is this happening every time you remove an entry? Any other details you can share?

**Comment by brainstuff** [comment]
Try keepass2adroid ... On Android Phones keepass2android is a better solution, it works with the same database ...  normally I do not like projects who are clones of existing software (like the clone of thunderbird by example) but keepass2android seems to work better as the android version of keepass.

**Comment by J-Jamet** [comment]
@brainstuff It doesn't help solve the problem. It's like you said to buy a 4x4 when you have a door problem with your city car ... People don't necessarily have the same needs.

@bpellin I think you can reproduce the problem only with databases created with KeePass 1.X (your V3 notation) on windows and not with KeePass 2.X. I don't know why, probably a problem in formatting and saving the data, maybe a byte size that is no longer the right one.

**Comment by sbhatarah** [comment]
@bpellin - I have the exact same issue as @smvicente with the same specs: using 'kdb' on pixel 2 XL. The issue only happens when deleting entries. Not sure what other information I can provide to help.

**Comment by bpellin** [comment]
Is there another keepass program you use to edit the same database? If so,
what version?

Are you using another program to browse to you file, or to sync the file
with different devices?

I'm looking for some common patterns, to help figure out how to reproduce
the problem.

**Comment by smvicente** [comment]
Hi @bpellin! thank you for your answer.

In all my tests, it only happens right after removing an entry using KeePassDroid on the Pixel 2XL. I also use KeePass X 0.4.X on Linux with the same kdb sometimes if it helps, but both open and edit the file just fine, except for this specific case on KeePassDroid.

**Comment by sbhatarah** [comment]
Hi @bpellin 

On my windows laptop I have "Keepass Password Safe 1.36 but I rarely use this. 

I normally only use KeePassDroid to edit the 'kdb' database, which I save locally on my phone's internal storage and then I just save backup copies to my laptop/Google drive as and when required. As mentioned the issue only occurs when deleting entries not when adding new ones. Also, I believe the issue only started happening after the most recent update.

I hope this helps. Let me know if you need any more info.

**Comment by RichardHaakma** [comment]
I want to report on this as well. I have a Pixel XL which is on LineageOS 17.1, which is is based on Android 10, if that makes a difference. I have recently had a corrupted v1 database similar to the above reports. So I restored a backup, verified that the backup was able to be opened. I increased the size of one entry, then closed and reopened the database. I then reduced the size of the same entry, closed and was not able to reopen the database.
So it is my theory that all I have to do is reduce the size of something. It is not necessary to remove an entry. Perhaps the code is not truncating the file properly when rewriting it smaller than before.
I am using the current version as of this date, 2.5.12.

Are more users using v2 database files now? Maybe working with v2 files are better tested. I note that KeepassX works with v2 files and this app does so maybe I will convert if it is generally recommended.
